codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ConnectionHandler.java

```diff
@@ -153,6 +134,10 @@ public void channelActive(ChannelHandlerContext ctx) throws Exception {
 
     @Override
     public void handlerAdded(ChannelHandlerContext ctx) throws Exception {
+        // Initialize the encoder and decoder.
+        encoder.lifecycleManager(this);
```

Comment:
@nmittler sry for being unclear. I'm wondering if we should check if the encoder and decoder already have a lifecycle manager set before we set them (also check to make sure the are the same after). Haven't thought about it too much and I would have to check if the http2connectionhandler has any limitations where it has to be the lifecyclemanager for the encoder/decoder that it uses. Not essential but wdyt?